test/Decimal.test.ts

```diff
@@ -333,16 +340,17 @@ describeNonPool("Decimal", () =>
 
   describe("DecimalUtils", () =>
   {
-    function bn18(x:Numberish): BigNumber {
+    function bn18(x:Numberish): bigint {
       const parts = x.toString().split('.');
-      const whole = BigNumber.from(parts[0]);
+      const whole = BigInt(parts[0]);
       if (parts.length === 1) // it was an integer, return it as-is
         return whole;
       // it's a decimal fraction, scale it up:
-      const sign = BigNumber.from(whole.isNegative() ? '-1' : '1');
+      const sign = BigInt(whole >= 0 ? '1' : '-1');
```

Comment:
I think it maybe safer to use the same type?

```suggestion
      const sign = BigInt(whole >= 0n ? '1' : '-1');
```

Reply:
We'll have to upgrade ES in another PR to get that feature